arangod/Aql/IResearchViewNode.cpp

```diff
@@ -1622,6 +1628,10 @@ void IResearchViewNode::doToVelocyPack(VPackBuilder& nodes,
   // volatility mask
   nodes.add(kNodeVolatilityParam, VPackValue(_volatilityMask));
 
+  if (_immutableParts != 0) {
+    nodes.add(kNodeViewImmutableParts, VPackValue{_immutableParts + 1U});
```

Comment:
why +1 here and -1 on deserialization?

Reply:
because uint_max is common case

Reply:
Please add a comment. Otherwise this part just looks weird.